### PR TITLE
test cases for parquetMetrics with multiple rowgroups

### PR DESCRIPTION
**Pull request by manishmalhotrawork**
this PR is for issue #132.
-- tested firstLevel fields and nestedStructure with multipleRowGroups.

@aokolnychyi can you please review.

### CODE REVIEW

File: parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java [hunk 1]
```diff
@@ -65,6 +70,180 @@
   private final GenericFixed fixed = new GenericData.Fixed(
       org.apache.avro.Schema.createFixed("fixedCol", null, null, 4),
       "abcd".getBytes(StandardCharsets.UTF_8));
+  @Test
```
Comment (rdblue) [inline parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java]: Nit: there should be an empty line between test cases and other code.

File: parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java [hunk 2]
```diff
@@ -65,6 +70,180 @@
...
+    secondRecord.put("timeCol", 3000L);
+    secondRecord.put("timestampCol", 1000L);
+    secondRecord.put("uuidCol", null);
+    secondRecord.put("fixedCol", fixed);
+    secondRecord.put("binaryCol", "W".getBytes());
+    records.add(secondRecord);
+
+    Record lastRecord = firstRecord;
+
+    for (int i = 0; i < minimumRowGroupRecordCount * 2; i++) {
```
Comment (rdblue) [inline parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java]: The process for creating the record set seems a bit too complicated to me. Why not base the records on `i` instead of adding 1 to the last record's value? This should also be a helper method.

File: parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java [hunk 3]
```diff
@@ -65,6 +70,180 @@
...
+      newRecord.put("fixedCol", fixed);
+      newRecord.put("binaryCol", "S".getBytes());
+      records.add(newRecord);
+      lastRecord = new Record(newRecord, false);
+    }
+
+    // create parquet file with multiple row groups. by using smaller number of bytes
+    File parquetFile = writeRecords(
+        schema,
+        ImmutableMap.of(PARQUET_ROW_GROUP_SIZE_BYTES, Integer.toString(minimumRowGroupRecordCount * Integer.BYTES)),
```
Comment (rdblue) [inline parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java]: Why is this size related to the number of records? What about just using a small number, like 1mb and a few thousand records?

**Comment by aokolnychyi** [comment]
Thanks for pinging me, @manishmalhotrawork. I don't have any other comments other than what @rdblue already mentioned. Thanks for working on this!

File: parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java (hunk 2 above)
Comment (manishmalhotrawork) [reply parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java]: @rdblue thanks for the review !

I have simplified the method to generate the records, will update the PR.
for helper method, as here Im also generating null values, for few columns.
Plus the values we are adding is depending on the schema.
So, not sure should be we have now, or if we have more places where its required then create a helper as well.
Please let me know what do you think.

File: parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java (hunk 3 above)
Comment (manishmalhotrawork) [reply parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java]: @rdblue thanks for the reviewing this !

I find it is interesting :)
Yes this value can be independent, but I kept it like this, to derive number of row groups based on the rows we added. ( if we see we added 200+ records, and we got 3 row groups)

Another point I find which is interesting is about the why records has to be > 100, if we want more than 1 row groups. 
Looks like 100 is minimum value for minNumberOfRecords in each row group.
So, we use  100+ records to be written, and give rowGroupSize = (100*1 column size).

details about this value : ( I believe you would be aware of, so may be can skip lengthy part :) )
If we see this existing [TestParquet](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/test/java/org/apache/iceberg/parquet/TestParquet.java#L85-L94)

It's using the 100 as the number of records, and if we run this test with < 100 records (say 80) , and even though the [here](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/test/java/org/apache/iceberg/parquet/TestParquet.java#L9)1 we pass say `20` still the it doesn't create more then 1 row groups.

what I understood, reason this [startRowGroup](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java#L177)

It calculates this value as 
`this.nextCheckRecordCount = min(max(recordCount / 2, 100), 10000);` 
so, even if the `recordCount < 100 ` the `nextCheckRecordCount` will be 100.
and this variable is used eventually when it writes records, from [checkSize()](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java#L150)  which is called from [add()](https://github.com/apache/incubator-iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java#L118).

File: parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java (hunk 2 above)
Comment (manishmalhotrawork) [reply parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java]: @rdblue thanks for reviewing !
Now the class `TestMetrics` has the test methods now.
So, will add these methods there.

File: parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java (hunk 3 above)
Comment (manishmalhotrawork) [reply parquet/src/test/java/org/apache/iceberg/parquet/TestParquetUtil.java]: also in the case of `testMetricsForTopLevelWithMultipleRG`, my understanding is.
it internally uses 
[ParquetWriter.java](https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java)
which uses [InternalParquetRecordWriter.java](https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordWriter.java#L45) which also has `MINIMUM_RECORD_COUNT_FOR_CHECK = 100` which is also making sure each row group will have minimum 100 records.

**Comment by manishmalhotrawork** [comment]
closing this one, as `TestParquetMetrics` is refactored.
opened [PR-365](https://github.com/apache/incubator-iceberg/pull/365)
